src/status_im/hardwallet/fx.cljs

```diff
@@ -95,3 +98,32 @@
  :send-transaction-with-signature
  (fn [{:keys [transaction signature on-completed]}]
    (status/send-transaction-with-signature transaction signature on-completed)))
+
+(re-frame/reg-fx
+ :hardwallet/persist-pairings
+ (fn [pairings]
+   (.. js-dependencies/async-storage
+       (setItem "status-keycard-pairings" (types/serialize pairings)))))
+
+(re-frame/reg-fx
+ :hardwallet/retrieve-pairings
+ (fn []
+   (when platform/android?
+     (.. js-dependencies/async-storage
+         (getItem "status-keycard-pairings")
+         (then #(re-frame/dispatch [:hardwallet.callback/on-retrieve-pairings-success
+                                    (types/deserialize %)]))))))
+
+(re-frame/reg-fx
+ :hardwallet/listen-to-hardware-back-button
+ (fn []
```

Comment:
```suggestion
 ;;NOTE: not done in view because effect should happen under different conditions and is not dependent on 
 ;;particular screen to be loaded. An fx is easier to re-use and test.
 (fn []
```